src/index.ts

```diff
@@ -283,37 +280,46 @@ export class Upload extends Pumpify {
   }
 
   private async startUploading() {
-    const reqOpts = {
-      method: 'PUT',
-      url: this.uri!,
-      headers: {
-        'Content-Range': 'bytes ' + this.offset + '-*/' + this.contentLength
-      }
-    };
-
     const bufferStream = this.bufferStream = new PassThrough();
     const offsetStream = this.offsetStream =
-        new PassThrough({transform: this.onChunk.bind(this)});
+        new Transform({transform: this.onChunk.bind(this)});
     const delayStream = new PassThrough();
+    const requestStreamEmbeddedStream = new PassThrough();
 
-    const requestStream = await this.getRequestStream(reqOpts);
-    this.setPipeline(bufferStream, offsetStream, requestStream, delayStream);
-
-    // wait for "complete" from request before letting the stream finish
     delayStream.on('prefinish', () => {
       this.cork();
     });
 
-    requestStream.on('complete', resp => {
-      if (resp.statusCode < 200 || resp.statusCode > 299) {
+    this.on('response', (resp: GaxiosResponse) => {
+      if (resp.data.error) {
+        this.destroy(resp.data.error);
+        return;
+      }
+
+      if (resp.status < 200 || resp.status > 299) {
         this.destroy(new Error('Upload failed'));
         return;
       }
 
-      this.emit('metadata', resp.body);
+      this.emit('metadata', resp.data);
       this.deleteConfig();
       this.uncork();
     });
+
+    this.setPipeline(bufferStream, offsetStream, delayStream);
+
+    this.pipe(requestStreamEmbeddedStream);
+
+    const reqOpts: GaxiosOptions = {
```

Comment:
I'm guessing you want to override `validateStatus` like the other requests:

`validateStatus: () => true`

Reply:
Done!